core/src/main/java/com/graphhopper/routing/util/spatialrules/countries/GermanySpatialRule.java

```diff
@@ -59,13 +59,10 @@ public double getMaxSpeed(RoadClass roadClass, TransportationMode transport, dou
             case PRIMARY:
                 return 100;
             case SECONDARY:
-                return 100;
             case TERTIARY:
-                return 100;
             case UNCLASSIFIED:
-                return 100;
             case RESIDENTIAL:
-                return 100;
+                return 50;
             case LIVING_STREET:
                 return 4;
```

Comment:
This does not look right. My understanding of the spatial rules was that they mirror the laws a country has regarding the maximum default speed. But 50km/h is only required inner city. Did you analyse the underlying cause you had to change this?

Reply:
The SpatialRule acts as a fallback here. We don't know if we're in urban area or not so we have to go with the more defensive assumption that we are.

Otherwise things like #2266 happen: The routing prefers an urban residential road as the maxspeed fallback is 100km/h  which is higher than the 70km/h of the B85. The routing acts accordingly and favors a detour.

Reply:
Hm, this sounds not the way to go. A spatial rule should stay like we defined and be identical to the local laws. If we mix in ETA improvements already here it won't work in the long term. So this PR is surely a step forward but we likely need to start with the ETA improvements before.

Reply:
> A spatial rule should stay like we defined and be identical to the local laws.

I agree but the law clearly differentiates between urban and non-urban speed limits. Using 100km/h as the fallback for all residential roads is a clear violation of the law for most urban areas.

Reply:
I see what you mean but then we need to expand the spatial rules to support inner and outer limits. But IMO this is a bigger undertaking that we need to plan a bit. E.g. the first step would be to identify from OSM and other sources where "inner city" is, from there we could populate an encoded value `inner_city` and finally apply the correct rural or urban spatial rule.

For the first step we should be able to use vector (define our own areas or grab data from OSM) and raster data (using e.g. satellite imagery).

Reply:
Right now, SpatialRules are in a Schrödinger's cat situation: they are both right and wrong depending on which road segment they handle. I agree that we should add the distinction to GH at some point in the future, but for now, defaulting to the lower boundaries is a workable solution.